freshmaker/pyxis.py

```diff
@@ -0,0 +1,205 @@
+
+        :param list index_images: list of index images
+        :param dict params: parameters to add to GET request
+        :return: bundle images per index image
+        :rtype: dict
+        """
+        bundles_per_index_image = {}
+        # we need 'bundle_path_digest' to find ContainerImage of that bundle
+        include_fields = self._default_include_fields + \
+                         ['data.channel_name', 'data.version',
+                          'data.related_images', 'data.bundle_path_digest']
+        for index_image in index_images:
+            path = index_image.get('path', '')
+            if not path:
+                continue
+            params.update({"filter": f"source_index_container_path=={path}",
+                           "include": ','.join(include_fields)})
+            bundles = self._pagination('operators/bundles', params)
+            bundles_per_index_image.update({path: bundles})
+
+        return bundles_per_index_image
+
+    def get_bundles_per_channel_per_index_image(self, index_images, params={}):
+        """
+        Get bundle images per channel per index image
+
+        :param list index_images: list of index images
+        :param dict params: parameters to add to GET request
+        :return: bundle images per channel per index image
+        :rtype: dict
+        """
+        bundles_per_index_image = \
+            self._get_bundles_per_index_image(index_images, params)
+
+        ret = {}
+        for index_image, bundles in bundles_per_index_image.items():
+            bundle_per_channel = {}
+            # get latest versions of bundle images per channel
+            for bundle in bundles:
+                channel = bundle.get('channel_name')
+                if channel in bundle_per_channel:
+                    new_version = bundle.get('version')
+                    old_version = bundle_per_channel[channel].get('version')
+                    if version.parse(new_version) > version.parse(old_version):
+                        bundle_per_channel[channel] = bundle
+                else:
+                    bundle_per_channel[channel] = bundle
+
+            ret[index_image] = bundle_per_channel
+        return ret
+
+    def get_digests_by_nvrs(self, nvrs, params={}):
+        """
+        Get images' digests by their nvrs
+
+        :param list nvrs: list of nvrs of ContainerImages to query Pyxis
+        :param dics params:
+        """
+        include_fields = self._default_include_fields + \
+                         ['data.brew', 'data.repositories.manifest_list_digest']
+        # TODO filter by architecture?
+        params.update({"include": ','.join(include_fields)})
+        images = []
+        for nvr in nvrs:
+            images = self._pagination(f'/images/nvr/{nvr}', params)
+
+        # get all manifest_list_digest of ContainerImages we got from Pyxis
+        digests = set()
+        for image in images:
+            if 'repositories' in image and \
```

Comment:
This will guard against an empty list for `image['repositories']`.
```suggestion
            if image.get('repositories') and \
```